include/roboteam_ai/stp/PlayEvaluator.h

```diff
@@ -0,0 +1,121 @@
+//
+// Created by maxl on 18-02-21.
+//
+
+#ifndef RTT_PLAYEVALUATOR_H
+#define RTT_PLAYEVALUATOR_H
+
+#include "world/World.hpp"
+
+namespace rtt::ai::stp{
+    enum class GlobalEvaluation{
+        /// Game States
+        BallPlacementThemGameState,
+        BallPlacementUsGameState,
+        FreeKickThemGameState,
+        FreeKickUsGameState,
+        HaltGameState,
+        KickOffThemGameState,
+        KickOffThemPrepareGameState,
+        KickOffUsGameState,
+        KickOffUsPrepareGameState,
+        NormalOrFreeKickUsGameState,
+        NormalPlayGameState,
+        PenaltyThemGameState,
+        PenaltyThemPrepareGameState,
+        PenaltyUsGameState,
+        PenaltyUsPrepareGameState,
+        StopGameState,
+        /// Global Evaluations
+        BallCloseToThem,
+        BallCloseToUs,
+        BallClosestToUs,
+        BallGotShot,
+        BallIsFree,
+        BallMovesSlow,
+        BallOnOurSide,
+        BallOnTheirSide,
+        BallShotOrCloseToThem,
+        DistanceFromBall,
+        FreedomOfRobots,
+        GoalVisionFromBall,
+        GoalVision,
+        NoGoalVisionFromBall,
+        WeHaveBall,
+        WeHaveMajority
+    };
+
+    class PlayEvaluator {
+    public:
+        struct PlayScoring {
+            uint8_t evaluationScore;
+            double weight;
+        };
+
+        /**
+         * Gets the score of a Global Evaluation, if it was not updated yet, update it before.
+         * @param evaluation that needs
+         * @return
+         */
+        uint8_t getGlobalEvaluation(GlobalEvaluation evaluation);
```

Comment:
why not by ref

Reply:
How so? I dont see how to implement a ref here

Reply:
```suggestion
        uint8_t getGlobalEvaluation(GlobalEvaluation &evaluation);
```
Im guessing John means this, and then do the same for everything connected to this function

Reply:
added it to this and more